pkg/prediction/dsp/prediction.go

```diff
@@ -30,9 +30,11 @@ const (
 
 type periodicSignalPrediction struct {
 	prediction.GenericPrediction
-	a           aggregateSignals
-	stopChMap   sync.Map
-	modelConfig config.AlgorithmModelConfig
+	a         aggregateSignals
+	stopChMap sync.Map
+	// record the query routine already started
+	queryRoutines sync.Map
```

Comment:
When does this map get set?

Reply:
Before we can not update the percentile algorithm config after start predicting, because general prediction will drop it  when call `WithQuery` when it is already registered, but we want to update the algorithm config of the metric dynamically.

so i moved the querySet to the algorithm internal, when user  call `WithQuery`, it can update the model config but do not started a  query go routine again

Reply:
> When does this map get set?

Before we can not update the percentile algorithm config after start predicting, because general prediction will drop it  when call `WithQuery` when it is already registered, but we want to update the algorithm config of the metric dynamically.

so i moved the querySet to the algorithm internal, when user  call `WithQuery`, it can update the model config but do not started a  query go routine again;

you can see the `line:166` is what i changed